sdk/couchbase/src/collection_ds.rs

```diff
@@ -0,0 +1,417 @@
+use crate::collection::Collection;
+use crate::options::kv_options::{MutateInOptions, StoreSemantics};
+use crate::subdoc::lookup_in_specs::LookupInSpec;
+use crate::subdoc::mutate_in_specs::MutateInSpec;
+use serde::de::DeserializeOwned;
+use serde::Serialize;
+use std::collections::HashMap;
+
+#[derive(Clone)]
+pub struct CouchbaseList<'a> {
+    pub collection: &'a Collection,
+    pub id: String,
+}
+
+impl Collection {
+    pub fn list(&self, id: impl Into<String>) -> CouchbaseList {
```

Comment:
It'd be a good idea for all of the constructor type functions to accept options blocks so that we can add functionality like tracing in the future.

Reply:
Done